pkg/hostagent/setup.go

```diff
@@ -306,21 +317,26 @@ func (agent *HostAgent) configureContainerIfaces(metadata *md.ContainerMetadata)
 		}
 		// if no mac is assigned, set it to the default Mac.
 		if len(iface.Mac) == 0 {
+			logger.Infof("No Mac assigned, assigning default")
 			iface.HostVethName, iface.Mac, err =
 				runSetupVeth(iface.Sandbox, iface.Name, agent.config.InterfaceMtu, nil)
 			if err != nil {
+				deallocIP(iface)
 				return nil, err
 			}
 		}
 
+		if len(iface.HostVethName) == 0 || len(iface.Mac) == 0 {
+			deallocIP(iface)
+			return nil, fmt.Errorf("Unable to Configure Container Interface")
```

Comment:
I think it will also help to log the values of len(iface.HostVethName) and len(iface.Mac) here.